spec/requests/metrics_spec.rb

```diff
@@ -3,25 +3,308 @@
 require 'spec_helper'
 
 RSpec.describe 'Metrics Endpoints', type: :request do
-  let(:admin) { create(:admin_user) }
-
   describe '#root' do
-    context 'no facets provided' do
-      before { get '/v3/metrics' }
+    context 'when no facets provided' do
+      before { get '/v3/metrics.json?' }
 
-      it 'returns user info' do
+      it 'returns error' do
         response_attributes = JSON.parse(response.body)
 
         expect(response_attributes).to eq({ 'errors' => 'facets parameter is required' })
       end
     end
+
+    context 'when facets length is more than MAX_FACETS' do
+      before do
+        params = { facets: (1..11).to_a.join(',') }.to_query
+        get "/v3/metrics.json?#{params}"
+      end
+
+      it 'returns error' do
+        response_attributes = JSON.parse(response.body)
+
+        expect(response_attributes).to eq({ 'errors' => 'Only up to 10 may be requested at once' })
+      end
+    end
+
+    # context 'when default metrics data for all facets is requested' do
```

Comment:
These specs are commented because the endpoint has logical bugs that needs to be fixed. When its fixed these specs should work.

Reply:
`xcontext` will skip the whole block if you don't want to comment them out

Reply:
Done!